src/domain/users/index.types.d.ts

```diff
@@ -45,4 +45,11 @@ type User = {
 interface IUsersRepository {
   findById(userId: UserId): Promise<User | RepositoryError>
   update(user: User): Promise<User | RepositoryError>
+  addContactByWalletName({
```

Comment:
This should not be on the repository - adding a contact is business logic.

Should be:
```
let user = UsersRepository.findByWalletName(walletName)
user.addContact(contact)
UsersRepository.update(user)
```

Similar to how we add `lastIp` in [this use-case](https://github.com/GaloyMoney/galoy/blob/main/src/app/users/index.ts#L51-L78)

Reply:
I missed this one, thanks!

Refactored now to place/implement this logic appropriately.